### PR TITLE
Create transcation type

### PR DESCRIPTION
**Pull request by BenjaminParment**
-> Tests needs a little fix

### CODE REVIEW

File: src/components/submissions/submissions-table/SubmissionsTable.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ export interface SubmissionRow {
 
 const cellRenderers: CellRenderers<SubmissionRow> = { errorRenderer: ErrorCell };
 
+/*ADD "tooltip: params => params.value" to each header for tooltips */
```
Comment (j-n-white) [inline src/components/submissions/submissions-table/SubmissionsTable.tsx]: I think you might as well do this as part of this change if it is needed to make the tooltips work

File: src/resources/svgs/InfoIcon.tsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { SVGProps } from "./svgProps";
 
 export class InfoIcon extends React.Component<SVGProps> {
     render = () => (
-        <svg {...this.props} viewBox="0 0 24 24">
+        <svg {...this.props} fill="#B3B3B3" viewBox="0 0 24 24">
```
Comment (j-n-white) [inline src/resources/svgs/InfoIcon.tsx]: Can we do this is CSS instead so we can link this to the appropriate variable

File: .env.development [hunk 3]
```diff
@@ -1,7 +1,7 @@
 PORT=30004
 REACT_APP_DYNAMIC_AUTH=false
 REACT_APP_COGNITO_REDIRECT_URL=http://localhost:30004
-REACT_APP_API_ENDPOINT=http://localhost/poc
+REACT_APP_API_ENDPOINT=http://localhost:3000
 REACT_APP_COGNITO_APP_CLIENT_ID=SampleClientId
-REACT_APP_COGNITO_ENDPOINT=http://localhost:30001
+REACT_APP_COGNITO_ENDPOINT=http://localhost:3000
```
Comment (j-n-white) [inline .env.development]: This will need reverting

File: src/resources/svgs/SvgStylings.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+@import "~shared/style/colors";
+
+.info-icon {
+    fill: $MID_FORGROUND_COLOR;
```
Comment (j-n-white) [inline src/resources/svgs/SvgStylings.scss]: Surely the icon should be agnostic towards its colour and the colour is defined wherever it is used

File: src/shared/utils/formatters.ts [hunk 5]
```diff
@@ -3,4 +3,6 @@ import moment from "moment";
 
 export const formatDate = (params: FormatterParams<unknown>) => params.value.format("D MMM YYYY");
 
+export const formatTooltipDate = (date: number) => moment(date).format("D MMM YYYY");
```
Comment (j-n-white) [inline src/shared/utils/formatters.ts]: Isn't the data infact a moment?
Then you can make the formatDate function call this. Also I'd call this function `formatDate` and the other function `formatTableDate`

File: .env.development (hunk 3 above)
Comment (BenjaminParment) [reply .env.development]: i keep forgetting this, damn

File: src/shared/utils/formatters.ts (hunk 5 above)
Comment (BenjaminParment) [reply src/shared/utils/formatters.ts]: will correct this now

File: src/components/reusables/table/headerFactory.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+export const createHeader = <T>(
+    headerName: string,
+    field: string,
+    valueFormatter?: (params: FormatterParams<T>) => string,
+    width?: number,
+    tooltip?: (params: ITooltipParams) => string,
+    cellRenderer?: string,
+): Column<T> => {
+    return {
```
Comment (j-n-white) [inline src/components/reusables/table/headerFactory.ts]: you don't need the return statement here

File: src/components/reusables/table/headerFactory.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { Column, FormatterParams } from "./Table";
+
+import { ITooltipParams } from "ag-grid-community";
+
+export const createHeader = <T>(
+    headerName: string,
+    field: string,
+    valueFormatter?: (params: FormatterParams<T>) => string,
+    width?: number,
+    tooltip?: (params: ITooltipParams) => string,
```
Comment (j-n-white) [inline src/components/reusables/table/headerFactory.ts]: put this as a defaulted parameter then you don't need to check for it being defined in the function body.
```suggestion
    tooltip: (params: ITooltipParams) => string = params => params.value,

```

File: src/shared/utils/formatters.test.ts [hunk 8]
```diff
@@ -26,8 +26,8 @@ describe("Given a call to formatDate", () => {
     });
 });
 
-describe("Given a call to formateDateTime", () => {
-    it("then the correctly formatted date and time is returned", () => {
-        expect(formatDateTime(moment("2020-01-02 13:00"))).toBe("2 Jan 2020 at 01:00pm");
+describe("Given a call to formatTooltipDate", () => {
+    it("then the correctly formatted date is returned", () => {
+        expect(formatTooltipDate(1578151853413)).toBe("4 Jan 2020");
```
Comment (j-n-white) [inline src/shared/utils/formatters.test.ts]: I think you deleted the wrong test

File: src/components/submissions/submissions-table/SubmissionsTable.tsx [hunk 9]
```diff
@@ -28,11 +28,11 @@ const cellRenderers: CellRenderers<SubmissionRow> = { errorRenderer: ErrorCell }
 
 const subbmissionsColumns: Column<SubmissionRow>[] = [
     createHeader("Batch ID", "batchId"),
-    createHeader("Created on", "createdOn", formatDate, undefined, formatDate),
-    createHeader("Product", "products", formatProductArray),
-    createHeader("Transactions", "transactions", displayIfValid),
-    createHeader("Tot. amount", "totalAmount", formatTotalAmount, 90),
-    createHeader("", "errorDisplay", undefined, 300, undefined, "errorRenderer"),
+    createHeader("Created on", "createdOn", formatDate, formatDate, undefined),
```
Comment (j-n-white) [inline src/components/submissions/submissions-table/SubmissionsTable.tsx]:
```suggestion
    createHeader("Created on", "createdOn", formatDate, formatDate),

```

**Pull request merged**